zip-0208.rst

```diff
@@ -0,0 +1,131 @@
+::
+
+  ZIP: 208
+  Title: Shorter Block Target Spacing
+  Author: Simon Liu <simon@z.cash>
+          Daira Hopwood <daira@z.cash>
+  Category: Consensus
+  Created: 2018-01-10
+  License: MIT
+
+
+Terminology
+===========
+
+The key words "MUST", "SHOULD", "SHOULD NOT", and "MAY" in this document are to be interpreted as
+described in RFC 2119. [#RFC2119]_
+
+The terms "block chain", "consensus rule change", "branch", and "network upgrade" are to be
+interpreted as defined in [#zip-0200]_.
+
+The term "block target spacing" means the time interval between blocks targeted by the difficulty
+adjustment algorithm in a given branch. It is normally measured in seconds.
+
+
+Abstract
+========
+
+This proposal specifies a change in the block target spacing, to take effect in the Blossom
+network upgrade [#zip-0206]_.
+
+The emission schedule of mined ZEC will be approximately the same in terms of time, but this
+requires the emission per block to be adjusted to take account of the changed block target spacing.
+
+
+Motivation
+==========
+
+- Faster finalization
+  TODO: does it actually provide this? Depends on dual PoW mechanism and recommended required confirmations.
+- Greater throughput of transactions in unit time
+
```

Comment:
We need to describe the measurements that are needed in order to decide on a block time (i.e. verification and propagation time for a worst-case block).

Reply:
Block Target Spacing
=================

To determine a value for target spacing which meets desired finalization and security requirements, it will be necessary to measure how long it takes a node to accept, process and propagate a block.

Test data for ProcessNewBlock() could be a block which exhibits:
  - worst-case block verification time (https://github.com/zcash/zcash/issues/3790)
  - maximum block size, currentlly 2MB
  - a single transaction spending as many Sapling spends as possible
  - each spend requires verifying RedDSA signature and Groth16 proof

Measure block propagation (related to https://github.com/zcash/zcash/issues/893) using nodes located around the world in a closed testnet.  Similar to https://scalingbitcoin.org/stanford2017/Day1/Stanford_2017.pptx.pdf

Reply:
Request: It occurs to me that the worst-case cases may already be intended benchmark tests. If we need benchmark vectors not already included in `zcashd`, let's add to the existing set.

Non-Blocking Wishlist: (TODO: File tickets for these.)

Less important for impacting consensus itself, but still very important for UX are metrics associated with bootstrapping, restarting, or maintaining a node:

- total time to `-reindex` at a given block height. 

  This may be difficult to model prior to deploying the change, but at the very least we should track this performance metric for the main chain over time.

  It seems likely to me that this change won't impact the function of reindex time versus number of blocks (ie: I don't see any dependence of reindex time on target time itself). If others agree that's a safe assumption, we can extrapolate from current data. For any extrapolation we must always anticipate new bottlenecks not captured by the training data.

- total time to sync a new node over the network.
- time (and other resources) to load the block index.
- same for the wallet (for a test-vector wallet profile).
- diskspace per block.
- all of the above parameters when `txindex=1`.